app/components/key_pair_generator_component.js

```diff
@@ -0,0 +1,40 @@
+import { trackEvent } from '@18f/identity-analytics';
+
+class KeyPairGeneratorElement extends HTMLElement {
+  connectedCallback() {
+    this.generateKeyPair().then(() => {
+      if (this.duration === undefined) {
+        this.duration = -1;
+      }
+      trackEvent('IdV: key pair generation', {
+        duration: this.duration.toFixed(0),
+        location: this.dataset.location,
+      });
+    });
+  }
```

Comment:
Continuing from my prior comment, an alternative way to address the TypeScript issue with potentially-undefiend `duration` is to avoid the side-effect and make the duration a return value of the function, since that's the value we're primarily concerned with anyways.

For example:

```ts
class KeyPairGeneratorElement extends HTMLElement {
  connectedCallback() {
    this.logDuration();
  }

  async logDuration() {
    const duration = await this.getGeneratedKeyPairDuration();
    trackEvent('IdV: key pair generation', {
      duration: this.duration.toFixed(0),
      location: this.dataset.location,
    });
  }

  async getGeneratedKeyPairDuration(): Promise<number> {
    // ...
    return t1 - t0;
  }
}
```

Reply:
Gotcha - this better represents what we're doing today, so I made the change in b2e883bcb